packages/mdc-textfield/helper-text/README.md

```diff
@@ -0,0 +1,133 @@
+<!--docs:
+title: "Text Field Helper Text"
+layout: detail
+section: components
+excerpt: "The helper text provides supplemental information and/or validation messages to users"
+iconId: text_field
+path: /catalog/input-controls/text-fields/helper-text/
+-->
+
+# Text Field Helper Text
+
+The helper text provides supplemental information and/or validation messages to users. It appears on input field focus and disappears on input field blur by default, or it can be persistent.
+
+## Design & API Documentation
+
+<ul class="icon-list">
+  <li class="icon-list-item icon-list-item--spec">
+    <a href="https://material.io/guidelines/components/text-fields.html#text-fields-layout">Material Design guidelines: Text Fields Layout</a>
+  </li>
+</ul>
+
+
+## Usage
+
+### Using helper text
+
+MDC Text Fields can include helper text that is useful for providing supplemental
+information to users, as well for validation messages (covered below).
+
+```html
+<div class="mdc-text-field">
+  <input type="text" id="username" class="mdc-text-field__input" aria-controls="username-helper-text">
+  <label for="username" class="mdc-text-field__label">Username</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+<p id="username-helper-text" class="mdc-text-field-helper-text" aria-hidden="true">
+  This will be displayed on your public profile
+</p>
+```
+
+Helper text appears on input field focus and disappears on input field blur by default when using
+the Text Field JS component.
+
+#### Persistent helper text
+
+If you'd like the helper text to always be visible, add the `mdc-text-field-helper-text--persistent` modifier class to the element.
+
+```html
+<div class="mdc-text-field">
+  <input type="email" id="email" class="mdc-text-field__input">
+  <label for="email" class="mdc-text-field__label">Email address</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+<p class="mdc-text-field-helper-text mdc-text-field-helper-text--persistent">
+  We will <em>never</em> share your email address with third parties
+</p>
+```
+
+#### Helper text and accessibility
+
+Note that in every example where the helper text is dependent on the state of the input element, we
+assign an id to the `mdc-text-field-helper-text` element and set that id to the value of the
+`aria-controls` attribute on the input element. We recommend doing this as well as it will help
+indicate to assistive devices that the display of the helper text is dependent on the interaction with
+the input element.
+
+When using our vanilla JS component, if it sees that the input element has an `aria-controls`
+attribute, it will look for an element with the id specified and treat it as the text field's help
+text element, taking care of adding/removing `aria-hidden` and other a11y attributes. This can also
+be done programmatically, which is described below.
```

Comment:
hmm its really unclear to me as a reader exactly what can be don programmatically...and it is not obvious the programmatic part is also the "MDCTextFieldHelperText API" section. Can you take another pass on this paragraph?